cpp/src/examples/equal_example.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
```

Comment:
I think this one may not need an example, I think we generally have lot of examples, but my opinion is we should have less examples. may be a better case for this one is a test case. Others may feel differently and I would like to hear.